src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml.cs

```diff
@@ -136,5 +139,25 @@ private void SetChecked(string[] tables)
             chkTables.ItemsSource = null;
             chkTables.ItemsSource = items;
         }
+
+        private async void TxtSearchTable_TextChanged(object sender, System.Windows.Controls.TextChangedEventArgs e)
+        {
+            var search = TxtSearchTable.Text;
+
+            await Task.Delay(500); // Add a delay (like a debounce) so that not every character change triggers a search
+            if (search != TxtSearchTable.Text)
+            {
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(search))
+            {
+                chkTables.ItemsSource = items;
+                return;
+            }
+
+            var filteredItems = items.Where(x => x.Label.ToUpper().Contains(TxtSearchTable.Text.ToUpper())).ToList();
+            chkTables.ItemsSource = filteredItems;
```

Comment:
Yes, you are right. It will not work. I will get on it. 
Thanks.

Reply:
Yes, I think you need to add a structure like List<Tuple<string, bool>> to keep track of the actual selected tables.

Reply:
Actually, I ran into a little odd behavior, at least odd for me (and seems like this is an expected behavior, considering the project is using `Xceed.Wpf.Toolkit` libraries). 

Every time a checkbox in the tables grid is selected, the `IsChecked` property of respective element in the list `items` get updated as well. Meaning, there is a one way binding from `chkTables.Items` to `items`.

I also looked into the documentation of `CheckBoxList` at:
https://github.com/xceedsoftware/wpftoolkit/wiki/CheckListBox

In that page, description for `SelectedMemberPath` *kind of* implies the behavior I mentioned.

To test this behavior, I even wrote unit test under `CheckListBoxTest` class. And the test passes. Since this test class was created is to test a behavior of an external library, this test will be removed in future commit. 

Also, I tested the generated visual studio extension, and things seem to work as expected.

@ErikEJ any thoughts?

Reply:
So the PR is ready now?

Reply:
It is ready as far as my testing goes. 
Actually, I was kinda waiting for your feedback. If I didn't misunderstand your last comment, you were going to pull my branch and do some testing.